File: app/src/main/java/mobi/lab/componentsdemo/MainActivity.java
package mobi.lab.componentsdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    private View itemBoundedView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        itemBoundedView = findViewById(R.id.item_bounded_view);
        itemBoundedView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, BoundedViewActivity.class));
            }
        });
    }
}

File: app/src/main/java/mobi/lab/componentsdemo/main/MainActivity.kt
package mobi.lab.componentsdemo.main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import mobi.lab.componentsdemo.common.BaseActivity

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)


    }

    companion object {
        fun getIntent(context: Context): Intent {
            return Intent(context, MainActivity::class.java)
        }
    }
}
